substratest/assets.py

```diff
@@ -370,7 +386,7 @@ class CompositeTraintuple(_Asset, _FutureMixin):
     log: str
     in_head_model: InModel = None
     in_trunk_model: InModel = None
-    out_head_model: OutCompositeModel = None
+    out_head_model: OutCompositeHeadModel = None
     out_trunk_model: OutCompositeModel = None
```

Comment:
The existing `OutCompositeModel` could be renamed `OutCompositeTrunkModel`? 

One other solution would have been to set the `OutModel.storage_address` field with a default `None` value (but the responses validation won't check if a "normal" OutModel has always a field `storage_address` set).

Reply:
Yes for the renaming ! I'm waiting for other review on the companion PR to see what will be the final result.

As we want to consider Head Model as private (like data samples) having a storage address at `None` won't be the behavior that we want. We don't want to expose any storage address for this asset.

Reply:
Done